desktop/libs/notebook/src/notebook/connectors/hiveserver2.py

```diff
@@ -181,16 +181,37 @@ def create_session(self, lang='hive', properties=None):
     application = 'beeswax' if lang == 'hive' or lang == 'llap' else lang
 
     if has_session_pool():
-      session = Session.objects.get_tez_session(self.user, application, MAX_NUMBER_OF_SESSIONS.get())
+      try:
+        session = Session.objects.get_tez_session(self.user, application, MAX_NUMBER_OF_SESSIONS.get())
+      except Exception as e:
+        LOG.exception('HA failover?')
+        if 'Connection refused' in str(e) or 'Name or service not known' in str(e):
+          reset_ha()
+          session = db.open_session(self.user)
+          session = Session.objects.get_tez_session(self.user, application, MAX_NUMBER_OF_SESSIONS.get())
     elif not has_multiple_sessions():
```

Comment:
Same

Reply:
Once the above is done will look at this one.

Reply:
this is also fixed, working on other test cases to address the failures with zk lookup.